File: ibm/src/dotnet/Factors/Factors.cs
using System;
using System.IO;
using System.Diagnostics;
using System.Collections.Generic;
using Newtonsoft.Json.Linq;

namespace Factors
{
    public class FactorsDotnet
    {
        public JObject Main(JObject args)
        {
            string machineId = File.ReadAllText("/sys/class/dmi/id/product_uuid");
            string instanceId = File.ReadAllText("/proc/self/cgroup");
            string cpuinfo = File.ReadAllText("/proc/cpuinfo");
            string meminfo = File.ReadAllText("/proc/meminfo");
            string uptime = File.ReadAllText("/proc/uptime");

            long n = 2688834647444046;

            if(args["n"] != null) {
                bool parseOk = Int64.TryParse(args["n"].ToString(), out n);
                if(!parseOk) {
                    n = 2688834647444046;
                }
            } else {
                n = 2688834647444046;
            }

            Stopwatch sw = new Stopwatch();
            sw.Start();
		        List<long> result = factorCalc(n);
            sw.Stop();
		        
            JObject message = new JObject();
            message.Add("success", new JValue(true));
            JObject payload = new JObject();
            payload.Add("test", new JValue("cpu test"));
            payload.Add("n", new JValue(n));
            payload.Add("result", JToken.FromObject(result));
            payload.Add("time", new JValue(sw.Elapsed.TotalMilliseconds));
            message.Add("payload", payload);
            JObject metrics = new JObject();
            metrics.Add("machineid", new JValue(machineId));
            metrics.Add("instanceid", new JValue(instanceId));
            metrics.Add("cpu", new JValue(cpuinfo));
            metrics.Add("mem", new JValue(meminfo));
            metrics.Add("uptime", new JValue(uptime));
            message.Add("metrics", metrics);

            return (message);
        }

        public static List<long> factorCalc(long num) {
          List<long> n_factors = new List<long>();
         
          for (long i = 1; i <= Math.Floor(Math.Sqrt(num)); i ++)
            if (num % i == 0) {
              n_factors.Add(i);
              if (num / i != i) {
                n_factors.Add(num / i);
              }
            }

          n_factors.Sort();

          return n_factors;
        }
    }
}

File: ibm/src/go/factors/factors.go
package main

import (
    "time"
    "math"
    "sort"
    "strconv"
    "io/ioutil"
    "log"
)

func Main(params map[string]interface{}) map[string]interface{} {

    buf1, err := ioutil.ReadFile("/proc/self/cgroup")
	if err != nil {
		log.Fatal(err)
    }
    instance_id := string(buf1)

    buf2, err := ioutil.ReadFile("/proc/cpuinfo")
	if err != nil {
		log.Fatal(err)
    }
    cpuinfo := string(buf2)

    buf3, err := ioutil.ReadFile("/proc/meminfo")
	if err != nil {
		log.Fatal(err)
    }
    meminfo := string(buf3)

    buf4, err := ioutil.ReadFile("/proc/uptime")
	if err != nil {
		log.Fatal(err)
    }
    uptime := string(buf4)

    buf5, err := ioutil.ReadFile("/sys/class/dmi/id/product_uuid")
	if err != nil {
		log.Fatal(err)
    }
    machine_id := string(buf5)

    n := 2688834647444046

    if _, ok := params["n"]; ok {
        n, err = strconv.Atoi(params["n"].(string))
        if err != nil {
            log.Fatal(err)
        }
    } else {
        n = 2688834647444046
    }

    start := time.Now()
    var result []int = factors(n)
    elapsed := time.Since(start)
    
    msg := map[string]interface{}{}
    msg["success"] = true
    msg["payload"] = map[string]interface{}{}
    msg["payload"].(map[string]interface{})["test"] = "cpu test"
    msg["payload"].(map[string]interface{})["n"] = n
    msg["payload"].(map[string]interface{})["result"] = result
    msg["payload"].(map[string]interface{})["time"] = int64(elapsed / time.Millisecond)
    msg["metrics"] = map[string]string{}
    msg["metrics"].(map[string]string)["machineid"] = machine_id
    msg["metrics"].(map[string]string)["instanceid"] = instance_id
    msg["metrics"].(map[string]string)["cpu"] = cpuinfo
    msg["metrics"].(map[string]string)["mem"] = meminfo
    msg["metrics"].(map[string]string)["uptime"] = uptime

    return msg

 }
	
func factors(num int) []int {

    var n_factors = make([]int, 0)

    for i := 1; i <= int(math.Floor(math.Sqrt(float64(num)))); i++ {
        if num % i == 0 {
            n_factors = append(n_factors, i)
            if num / i != i {
                n_factors = append(n_factors, num / i)
            }
        }
    }

    sort.Ints(n_factors)

    return n_factors

}
